include/libvfio-user.h

```diff
@@ -436,31 +436,17 @@ void
 vfu_destroy_ctx(vfu_ctx_t *vfu_ctx);
 
 /**
- * Once the vfu_ctx is configured vfu_ctx_drive() drives it. This function waits
- * for commands coming from the client, and processes them in a loop.
- *
- * @vfu_ctx: the libvfio-user context to drive
- *
- * @returns 0 on success, -errno on failure.
- */
-int
-vfu_ctx_drive(vfu_ctx_t *vfu_ctx);
-
-/**
- * Polls, without blocking, an vfu_ctx. This is an alternative to using
- * a thread and making a blocking call to vfu_ctx_drive(). Instead, the
- * application can periodically poll the context directly from one of
- * its own threads.
- *
- * This is only allowed when LIBVFIO_USER_FLAG_ATTACH_NB is specified during
- * creation.
+ * Polls the vfu_ctx and processes the command recieved from client.
```

Comment:
I think the documentation isn't clear enough. We must clearly state that in blocking mode `vfu_ctx_run` will continue processing commands indefinitely, blocking until new requests are available, and only exits in case of error or if the client disconnects. In non-blocking mode the function will process one request if it's available, otherwise it immediatelly returns and the caller is responsible for periodically calling this function.

Reply:
Sure